streaming/base/dataset.py

```diff
@@ -308,6 +308,10 @@ class StreamingDataset(Array, IterableDataset):
         allow_unsafe_types (bool): If a shard contains Pickle, which allows arbitrary code
             execution during deserialization, whether to keep going if ``True`` or raise an error
             if ``False``. Defaults to ``False``.
+        world (World, optional): Override the world state (otherwise it is detected from standard
+            env vars). Defaults to ``None``.
+        tensor_parallelism (int, optional): Tensor parallelism to apply to the given or detected
```

Comment:
Does this mean that the same sample is provided to multiple (consecutive) GPUs?
ie do all gpus in the `world` get the same sample?
Can this be documented so the user knows?

Reply:
or is it that samples across worlds are repeated?
ie rankN in all worlds produces the same sample

Reply:
> Does this mean that the same sample is provided to multiple (consecutive) GPUs?

Yes

>  ie do all gpus in the world get the same sample?

Nah, just all (`tensor_parallelism`)-long shingles of GPUs

> or is it that samples across worlds are repeated?

Basically `World` is just Streaming's wrapper around node/rank env vars and `torch.utils.data.get_worker_info()`, telling you which one of those you are, there's just one of them, basically.